packages/allow-scripts/src/config.js

```diff
@@ -0,0 +1,358 @@
+      // if (err.code === 'ENOENT' && branchIsOptional) {
+      //   continue
+      // }
+      throw err
+    }
+    const depScripts = depPackageJson.scripts || {}
+    const lifeCycleScripts = ['preinstall', 'install', 'postinstall'].filter(
+      (name) => Object.prototype.hasOwnProperty.call(depScripts, name)
+    )
+
+    if (
+      !lifeCycleScripts.includes('preinstall') &&
+      !lifeCycleScripts.includes('install') &&
+      existsSync(path.join(filePath, 'binding.gyp'))
+    ) {
+      lifeCycleScripts.unshift('install')
+      depScripts.install = 'node-gyp rebuild'
+    }
+
+    if (lifeCycleScripts.length) {
+      /**
+       * @type {{
+       *   canonicalName: string
+       *   path: string
+       *   scripts: LavamoatPackageJson['scripts']
+       * }[]}
+       */
+      const collection = packagesWithScriptsLifecycle.get(canonicalName) || []
+      collection.push({
+        canonicalName,
+        path: filePath,
+        scripts: depScripts,
+      })
+      packagesWithScriptsLifecycle.set(canonicalName, collection)
+    }
+
+    if (FEATURE.bins && depPackageJson.bin) {
+      const binsList = /** @type {[string, string][]} */ (
+        Object.entries(normalizeBin(depPackageJson)?.bin || {})
+      )
+
+      binsList.forEach(([name, link]) => {
+        /** @type {BinInfo[]} */
+        const collection = binCandidates.get(name) || []
+        if (collection.length === 0) {
+          binCandidates.set(name, collection)
+        }
+        collection.push({
+          // canonical name for a direct dependency is just dependency name
+          isDirect: directDeps.has(canonicalName),
+          bin: name,
+          path: filePath,
+          link,
+          fullLinkPath: path.relative(rootDir, path.join(filePath, link)),
+          canonicalName,
+        })
+      })
+    }
+  }
+
+  const lavamoatConfig = packageJson.lavamoat || {}
+
+  const configs = {
+    lifecycle: indexLifecycleConfiguration({
+      packagesWithScripts: packagesWithScriptsLifecycle,
+      allowConfig: lavamoatConfig.allowScripts,
+    }),
+    bin: indexBinsConfiguration({
+      binCandidates,
+      allowConfig: lavamoatConfig.allowBins,
+    }),
+  }
+
+  const somePoliciesAreMissing = !!(
+    configs.lifecycle.missingPolicies.length ||
+    configs.bin.somePoliciesAreMissing
+  )
+
+  return {
+    packageJson,
+    configs,
+    somePoliciesAreMissing,
+    canonicalNamesByPath,
+  }
+}
+
+/**
+ * @typedef GetOptionsForBinOpts
+ * @property {string} rootDir
+ * @property {string} name
+ */
+
+/**
+ * @param {GetOptionsForBinOpts} param0
+ * @returns {Promise<BinInfo[] | undefined>}
+ */
+async function getOptionsForBin({ rootDir, name }) {
+  const {
+    configs: {
+      bin: { binCandidates },
+    },
+  } = await loadAllPackageConfigurations({ rootDir })
+
+  return binCandidates.get(name)
+}
+
+/**
+ * @typedef SetDefaultConfigurationOpts
+ * @property {string} rootDir
+ */
```

Comment:
same idea as re: `GetOptionsForBinOpts`

Reply:
Lets do actual cleanup and renaming separately from shuffling things around?